### PR TITLE
ITSADSSD-22844 Tabs uplift

### PR DESCRIPTION
**Pull request by uqmchu**

### CODE REVIEW

**Review by Bcrunk — APPROVED** [review]

**Comment by Bcrunk** [comment]
What changes are still left?

**Comment by uqmchu** [comment]
This PR is still in draft mode. But since you are reviewing it, can you please revise the SCSS and the use of selectors as the my don’t comply with our chosen style.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

File: packages/storybook-html/src/components/tabs/tabs.docs.mdx [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+  <Story id="components-tabs--tabs-reversed" />
+</Canvas>
+
+```html
+  <div class="tabs tabs--outline tabs--reversed">
+    ...
+  </div>
+```
+
+## How and when to use
```
Comment (uqmchu) [inline packages/storybook-html/src/components/tabs/tabs.docs.mdx:119]: These seem like generic tabs recommendations and might cover more cases than what we would like to recommend. For example, should we be telling people that they can use icon-only tabs at all? Does it fit into our UX strategy? Has this been signed-off by CX? @Bcrunk any thoughts?

File: packages/tabs/src/scss/_component.scss [hunk 2]
```diff
@@ -2,218 +2,214 @@
...
-.tabs-section {
-  // Tabs - default
-  .tabs {
+  * {
+    box-sizing: border-box;
+  }
+
+  // Tab list
+  //&__tab-list,
+  [role="tablist"] {
```
Comment (uqmchu) [inline packages/tabs/src/scss/_component.scss:13]: Use BEM as per our CSS standards.

File: packages/tabs/src/scss/_component.scss [hunk 3]
```diff
@@ -2,218 +2,214 @@
...
+    &:last-child {
+      margin-right: core.$space-none;
     }
 
-    // ***********************************
-    // Tabs - fixed
-    &--fixed {
-      position: fixed;
-      top: 0;
+    &:before {
```
Comment (uqmchu) [inline packages/tabs/src/scss/_component.scss:61]: Pseudo-elements should be written using a double colon (::). E.g. `.some-class::before, .some-class::after etc.` Some reading: https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-elements. @Bcrunk do you agree?

File: packages/storybook-html/src/components/tabs/tabs.stories.js [hunk 4]
```diff
@@ -1,19 +1,212 @@
...
+    <div class="tabs__tab-panel" role="tabpanel" id="tab23-panel" aria-labelledby="tab23" tabindex="0" hidden="">
+      <p>Tab3 panel text</p>
+    </div>
+  </div>
+  `;
+};
+
+tabsMinimal.storyName = 'Minimal';
+
+export const tabsLined = () => {
```
Comment (uqmchu) [inline packages/storybook-html/src/components/tabs/tabs.stories.js:78]: Active tab text doesn't line up with the inactive ones: 
Screen Shot 2021-02-15 at 12 20 58 pm

File: packages/tabs/src/scss/_global.scss [hunk 5]
```diff
@@ -5,3 +5,5 @@ $panel-border: core.$uq-neutral-light60; //e1e1e1;
 $active-color: core.$uq-yellow; //ffa200;
```
Comment (uqmchu) [inline packages/tabs/src/scss/_global.scss:5]: Names can be more consistent.

File: packages/tabs/src/scss/_component.scss [hunk 6]
```diff
@@ -2,218 +2,214 @@
...
+      font-weight: core.$font-weight-bold;
     }
 
-    // ***********************************
-    // Tabs - three
-    &--three {
-      .tab-title {
-        width: 33.33%;
-      }
+    > span {
```
Comment (uqmchu) [inline packages/tabs/src/scss/_component.scss:79]: Can you give this element a BEM class rather than target the `span`?

**Review by uqmchu — COMMENTED** [review]
@uqtgeor4 Please see enclosed comments.

**Pull request merged**